### DISCUSSION TITLE
Hyperscaler like SR-IOV VF

### DISCUSSION THREAD

**Discussion by koolipaasha** [telekom/das-schiff#18] [Q&A] [answered]
as i understand from one of your networking sessions ,your network architecture tries to bring Hyperscaler like CNI features to on prem but using FRR ,Calico ,Coil etc ie Routable Pod IPs.

Have you thought about or considering the same flexibility for SR-IOV VFs,which is a big challenge in Telco and often requires security exceptions because of the lack of Network Policies.I hope this gets a reply and dont get ignored like the last question :)

**Comment by Cellebyte** [comment, marked as answer]
Hey and good evening,

Hyperscaler-like SRIOV comes with specialized hardware and a lot of burden.
We are currently testing Bluefield cards  from Nvidia because they can support such use-case by using an arm driven Linux kernel to passthrough SRIOV into the correct Overlay.
From a platform perspective it is not very well thought through and has still a lot of missing features and limitations which need to be catched and addressed.
A more general approach would be the current kernel development around [AF_XDP](http://vger.kernel.org/lpc_net2018_talks/lpc18_paper_af_xdp_perf-v2.pdf).
They created a new capability called  CAP_BPF, to separate bpf functionaliy from CAP_SYS_ADMIN.

This year RedHat also showed the following presentation on FOSDEM 2023 regarding a POC named [CNDP](https://fosdem.org/2023/schedule/event/hybrid_netstack/).

And Cilium had also very good talks regarding the future of eBPF. 
* [meta-netdevices](https://fosdem.org/2023/schedule/event/meta_netdevices/)
* [cni-as-a-servicemesh](https://fosdem.org/2023/schedule/event/network_service_mesh/)


Also projectcalico showcased some approach.
* [calico-pod-multihoming](https://fosdem.org/2023/schedule/event/network_pods_to_multiple_networks/)

So our current approach does not deal with SRIOV, because we do not want to ride a dead horse for too long.
Hopefully this answer is sufficient for you.

**Reply by koolipaasha** [reply]
Thanks alot @Cellebyte for your quick detailed response and the links.Very helpful.Hopefully at this odd time you are not troubleshooting an Overlay mapping issue :)
I have looked at CNDP before ( i thought it was an Intel driven initiative ),its really promising but none of the vendors i have spoken to showed any interest in moving from userplane+VF approach (which works for them) to the new framework. Hopefully with native sub function support in Linux and with more operator demand they will consider CNDP over VF+DPDK.
